web/src/test/java/se/inera/intyg/webcert/web/service/facade/impl/GetUserResourceLinksImplTest.java

```diff
@@ -143,6 +146,39 @@ void shallNotIncludeSignedCertificatesListIfUserIsNotDoctor() {
             final var actualLinks = getUserResourceLinks.get(user);
             ResourceLinkFacadeTestHelper.assertExclude(actualLinks, ResourceLinkTypeDTO.ACCESS_SIGNED_CERTIFICATES_LIST);
         }
+
+        @Test
+        void shallNotIncludeChooseUnitIfOriginIsUthopp() {
+            final var user = getUserWithOrigin("UTHOPP");
+            final var actualLinks = getUserResourceLinks.get(user);
+            ResourceLinkFacadeTestHelper.assertExclude(actualLinks, ResourceLinkTypeDTO.CHOOSE_UNIT);
+        }
+
+        @Test
+        void shallNotIncludeChooseUnitIfOriginIsDjupintegration() {
+            final var user = getUserWithOrigin("DJUPINTEGRATION");
+            final var actualLinks = getUserResourceLinks.get(user);
+            ResourceLinkFacadeTestHelper.assertExclude(actualLinks, ResourceLinkTypeDTO.CHOOSE_UNIT);
+        }
+
+        @Test
+        void shallIncludeChooseUnitIfOriginIsNormalAndHasLoggedInUnit() {
```

Comment:
If I'm not wrong the user should get the resource link if she has not a set unit. If that is the case then this test should fail because if the user has a set unit the resource link should not be included. The other two tests are not useful since if the user has a set unit the resource link is gonna be excluded no matter what origin. You should do tests checking that a user with no unit doesn't have the resource link if the wrong origin, that the a user with the correct origin and no unit has the resource link, and that a user with the correct origin and a set user doesn't have the resource link.